ts/parts-more/BubbleSeries.ts

```diff
@@ -448,7 +453,8 @@ seriesType<Highcharts.BubbleSeriesOptions>('bubble', 'scatter', {
         zMax: number,
         minSize: number,
         maxSize: number,
-        value: number
+        value: number,
```

Comment:
@sebastianbochan Please see https://github.com/highcharts/highcharts/pull/11400#discussion_r304253652

If yData and zData can be `Array<number|null|undefined>`, then `getRadius` has to support `number|null|undefined` types for `value` and `yValue` as well.

We should avoid casting to `any` as we at some point will have to remove all `any`'s when we switch to strict mode in TypeScript.

Reply:
I am not sure if changes to `yData` imply a change for `getRadius`, but I agree, that we should check for defined types as much as possible to avoid `any` casts.

Reply:
Ok but when I use value as <number|null|undefined> then the problem is here:

```
if (yValue === null || value === null) {
            radius = null;
        // Issue #4419 - if value is less than zMin, push a radius that's
        // always smaller than the minimum size
        } else if (value < zMin) {
            radius = minSize / 2 - 1;
        } else {
            // Relative size, a number between 0 and 1
            pos = zRange > 0 ? (value - zMin) / zRange : 0.5;

            if (sizeByArea && pos >= 0) {
                pos = Math.sqrt(pos);
            }
            radius = Math.ceil(minSize + pos * (maxSize - minSize)) / 2;
        }
```

Lets analyze the second IF and ELSE. Linter blocks me because of "value can be undefined", but in JS it works properly. 

Morever other parametrs like `minSize` and `maxSize` can be declared as `any`. 

@bre1470 what do you think? Should we fix that or leave like it is?

Reply:
@sebastianbochan  What should `getRadius` return if `value=undefined`?
My first thought was that it could either return `null` or maybe `0`.

If `null` is the correct return value, then this is solved by doing:
```
if (yValue === null || value === null || value === undefined) {
  radius = null;
} else if ...
```

Reply:
Completely no.

If `value` is `undefined` it means that only z param is not defined (by user), but bubble should be visible because y value is declared. This is the title of bug.

Reply:
So what value should it return in this case? 🤔 Currently it will return `NaN` which although works somehow, I believe it is not a good return value.

Reply:
Maybe the condition has to be more specific. Something like `(value === void 0 /* undefined */ && !U.defined(yValue))`?

Reply:
@jon-a-nygaard 
Lets analyze all step by step.

When `value` is undefined the `zRange` is -Infinity. In this condition: ` pos = zRange > 0 ? (value - zMin) / zRange : 0.5;` value is not used and `pos = 0.5`. As a result we do not have NaN anywhere and radius is calculated correctly.

Do you agree with me?

Reply:
Sorry @sebastianbochan, but I do not. It may end up well in the case you are testing, but `getRadius` does not cover all possible situations at the moment.

Please see the following example of how `radius` could end up as `NaN`, open the console to view steps and values: https://jsfiddle.net/jon_a_nygaard/wut3moxz/

This is also the reason why TypeScript complains, because we are doing math operations on variables that do not have the type `number`.

But if I understand your last comment correctly, then maybe we could say that if `value===undefined` then `pos=0.5` to achieve the desired result?